conn.go

```diff
@@ -640,6 +646,14 @@ func (w *writeCoalescer) flush() {
 }
 
 func (w *writeCoalescer) Write(p []byte) (int, error) {
+	// TODO: use atomics for this?
+	w.fcond.L.Lock()
```

Comment:
I think it would be pretty safe here to just mark it as running and broadcasting without checking the current status.